tool/healthz/main.go

```diff
@@ -43,7 +44,13 @@ func main() {
 	s3AccessKeyID := flag.String("access-key-id", "", "S3 access key")
 	s3SecretAccessKey := flag.String("secret-access-key", "", "S3 secret key")
 	s3Endpoint := flag.String("endpoint", defaultEndpoint, "S3 endpoint address")
-	s3Bucket := flag.String("bucket", defaultBucket, "S3 Bucket name")
+	// make the bucket name unique, based on the hostname,
+	// to avoid collisions in multi-nodes clusters
+	hostname, err := os.Hostname()
+	if err != nil {
+		log.Fatalf("failed to retrieve pod hostname: %v", err)
+	}
+	s3Bucket := flag.String("bucket"+hostname, defaultBucket, "S3 Bucket name")
```

Comment:
I recommend here to delete flag s3Bucket. Make constant like `s3BucketPrefix` and create full name of the bucket with `fmt.Sprintf("%s-%s", s3BucketPrefix, hostname)`. I think configuring bucket flag via cli variable do not make sense more.